File: src/main/java/banane/io/pdb/security/SecurityServiceImpl.java
package banane.io.pdb.security;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.stereotype.Service;

import banane.io.pdb.model.User;
import banane.io.pdb.service.UserService;

@Service
public class SecurityServiceImpl implements SecurityService {

    @Autowired
    private AuthenticationManager authenticationManager;

    @Autowired
    private UserDetailsService userDetailsService;

    @Autowired
    private UserService userService;

    private static final Logger logger = LoggerFactory.getLogger(SecurityServiceImpl.class);

    @Override
    public String findLoggedInUsername() {
        logger.debug("Finding logged in user");
        Object userDetails = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        if (userDetails instanceof UserDetails) {
            final String username = ((UserDetails) userDetails).getUsername();
            logger.debug("Find logged in user with username : {}", username);
            return username;
        }
        logger.debug("No user found");
        return null;
    }

    @Override
    public User findLoggedInUser() {
        logger.debug("Trying to find current logged in user");
        return userService.findByUsername(findLoggedInUsername()).orElseThrow(() -> new IllegalStateException("A logged in user should have a User of the same name"));
    }

    @Override
    public void autologin(String username, String password) {
        UserDetails userDetails = userDetailsService.loadUserByUsername(username);
        UsernamePasswordAuthenticationToken usernamePasswordAuthenticationToken = new UsernamePasswordAuthenticationToken(userDetails, password, userDetails.getAuthorities());

        authenticationManager.authenticate(usernamePasswordAuthenticationToken);

        if (usernamePasswordAuthenticationToken.isAuthenticated()) {
            SecurityContextHolder.getContext().setAuthentication(usernamePasswordAuthenticationToken);
            logger.debug("Auto login {} successfully!", username);
        }
    }
}

File: src/main/java/banane/io/pdb/security/SecurityServiceImpl.kt
package banane.io.pdb.security

import banane.io.pdb.model.User
import banane.io.pdb.service.UserService
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.stereotype.Service

@Service
open class SecurityServiceImpl : SecurityService {
    @Autowired
    private val authenticationManager: AuthenticationManager? = null
    @Autowired
    private val userDetailsService: UserDetailsService? = null
    @Autowired
    private val userService: UserService? = null

    override fun findLoggedInUsername(): String? {
        logger.debug("Finding logged in user")
        val userDetails = SecurityContextHolder.getContext().authentication.principal
        if (userDetails is UserDetails) {
            val username = userDetails.username
            logger.debug("Find logged in user with username : {}", username)
            return username
        }
        logger.debug("No user found")
        return null
    }

    override fun findLoggedInUser(): User? {
        logger.debug("Trying to find current logged in user")
        return userService!!.findByUsername(findLoggedInUsername())!!.orElseThrow { IllegalStateException("A logged in user should have a User of the same name") }
    }

    override fun autologin(username: String?, password: String?) {
        val userDetails = userDetailsService!!.loadUserByUsername(username)
        val usernamePasswordAuthenticationToken = UsernamePasswordAuthenticationToken(userDetails, password, userDetails.authorities)
        authenticationManager!!.authenticate(usernamePasswordAuthenticationToken)
        if (usernamePasswordAuthenticationToken.isAuthenticated) {
            SecurityContextHolder.getContext().authentication = usernamePasswordAuthenticationToken
            logger.debug("Auto login {} successfully!", username)
        }
    }

    companion object {
        private val logger = LoggerFactory.getLogger(SecurityServiceImpl::class.java)
    }
}